core/src/main/java/discord4j/core/object/Template.java

```diff
@@ -0,0 +1,186 @@
+    }
+
+    /**
+     * Gets the amount of times the template has been used.
+     *
+     * @return The template usage count.
+     */
+    public final int getUsageCount() {
+        return data.usageCount();
+    }
+
+    /**
+     * Gets the id of the creator of this template.
+     *
+     * @return The creator id.
+     */
+    public final String getCreatorId() {
+        return data.creatorId();
+    }
+
+    /**
+     * Gets the creator of this template.
+     *
+     * @return The creator.
+     */
+    public final User getCreator() {
+        return new User(gateway, data.creator());
+    }
+
+    /**
+     * Gets the timestamp the template was created.
+     *
+     * @return The timestamp.
+     */
+    public final String getCreatedAt() {
+        return data.createdAt();
+    }
+
+    /**
+     * Gets the timestamp the template was updated.
+     *
+     * @return The timestamp.
+     */
+    public final String getUpdatedAt() {
+        return data.updatedAt();
+    }
+
+    /**
+     * Gets the guild snapshot of the template.
+     *
+     * @return The guild snapshot.
+     */
+    public final SerializedSourceGuildData getSourceGuild() {
+        return data.serializedSourceGuild();
+    }
+
+    /**
+     * Creates a new guild from a template. Fires Guild Create Gateway event.
+     *
+     * @return the guild object
+     */
+    public final Mono<Guild> createGuild(final Consumer<? super TemplateCreateGuildSpec> spec) {
+        return Mono.defer(() -> {
+            TemplateCreateGuildSpec mutatedSpec = new TemplateCreateGuildSpec();
+            spec.accept(mutatedSpec);
+            return gateway.getRestClient().getTemplateService().createGuild(getCode(), mutatedSpec.asRequest(), mutatedSpec.getReason());
+        }).map(data -> new Guild(gateway, data));
+    }
+
+    /**
+     * Requests to sync the current template with the guild.
+     */
+    public final Mono<Void> sync(long guildId) {
+        return gateway.getRestClient().getTemplateService()
+            .syncTemplate(guildId, getCode())
+            .then();
+    }
+
+    public final Mono<Void> sync() {
+        return this.sync(this.getGuildId());
+    }
+
+    /**
+     * Requests to edit this guild template.
+     *
+     * @param spec A {@link Consumer} that provides a "blank" {@link GuildTemplateEditSpec} to be operated on.
+     * @return A {@link Mono} where, upon successful completion, emits the edited {@link Template}. If an error is
+     * received, it is emitted through the {@code Mono}.
+     */
+    public final Mono<Template> edit(final Consumer<? super GuildTemplateEditSpec> spec) {
+        return Mono.defer(() -> {
+            GuildTemplateEditSpec mutatedSpec = new GuildTemplateEditSpec();
+            spec.accept(mutatedSpec);
+            return gateway.getRestClient().getTemplateService().modifyTemplate(getGuildId(), getCode(), mutatedSpec.asRequest(), mutatedSpec.getReason());
+        }).map(data -> new Template(gateway, data));
```

Comment:
Please use the formatting from other Mono.defer + spec methods.

Reply:
changed formatting